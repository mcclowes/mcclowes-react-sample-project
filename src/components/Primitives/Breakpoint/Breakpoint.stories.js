import React from "react";
import Breakpoint from "./index";

export default { title: "Breakpoint" };

export const alwaysHidden = () => {
  return (
    <div>
      Should never render
      <br />
      <Breakpoint>Some text</Breakpoint>
    </div>
  );
};

export const mobile = () => {
  return (
    <div>
      Should render at mobile sizes
      <br />
      <Breakpoint sizes={[true, true, false, false, false]}>
        Some text
      </Breakpoint>
    </div>
  );
};

export const desktop = () => {
  return (
    <div>
      Should render at desktop sizes
      <br />
      <Breakpoint sizes={[false, false]}>Some text</Breakpoint>
    </div>
  );
};

export const alwaysShowing = () => {
  return (
    <div>
      Should render always
      <br />
      <Breakpoint>Some text</Breakpoint>
    </div>
  );
};
