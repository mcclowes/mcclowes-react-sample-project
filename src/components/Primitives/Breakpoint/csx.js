import styled from "styled-components/macro";
import { breakpoint, breakpointSizes } from "../../../utils/styles/breakpoints";
import PropTypes from "prop-types";

const createBreakpointRule = (size, i) => {
  if (size) return "";

  return `
    ${breakpoint(breakpointSizes[i], "only")} {
      display: none;
    }
  `;
};

export const Breakpoint = styled.div`
  ${({ sizes }) =>
    sizes && sizes.map((size, i) => createBreakpointRule(size, i))};
`;

Breakpoint.propTypes = {
  sizes: PropTypes.arrayOf(PropTypes.bool),
};
