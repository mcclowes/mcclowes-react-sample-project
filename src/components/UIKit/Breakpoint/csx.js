import styled from "styled-components/macro";
import { styles } from "../../../utils";

const breakpointSizes = ["xSmall", "small", "medium", "large", "xLarge"];

export const Breakpoint = styled.div`
  ${(props) =>
    props.sizes &&
    props.sizes.map(
      (size, i) =>
        !size &&
        styles.breakpoints.breakpoint(
          breakpointSizes[i],
          "only",
          `
                  display: none;
                `
        )
    )};
`;
