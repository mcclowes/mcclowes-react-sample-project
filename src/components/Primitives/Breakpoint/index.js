import { Breakpoint } from "./csx";

export default Breakpoint;
