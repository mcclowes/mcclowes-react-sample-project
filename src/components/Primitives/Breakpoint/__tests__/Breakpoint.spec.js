import Breakpoint from "../index";
import React from "react";
import { shallow } from "enzyme";

let wrapper;

describe("Breakpoint", () => {
  it("renders component", () => {
    wrapper = shallow(<Breakpoint>Some text</Breakpoint>);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders component with sizes", () => {
    wrapper = shallow(
      <Breakpoint sizes={[true, true, false, false, false]}>
        Some text
      </Breakpoint>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
