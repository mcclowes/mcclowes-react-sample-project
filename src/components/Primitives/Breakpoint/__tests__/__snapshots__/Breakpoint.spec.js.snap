// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breakpoint renders component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "csx__Breakpoint-sc-16ry0sm-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          ";",
        ],
      },
      "displayName": "csx__Breakpoint",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "sizes": [Function],
      },
      "render": [Function],
      "styledComponentId": "csx__Breakpoint-sc-16ry0sm-0",
      "target": "div",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Some text
</StyledComponent>
`;

exports[`Breakpoint renders component with sizes 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "csx__Breakpoint-sc-16ry0sm-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          ";",
        ],
      },
      "displayName": "csx__Breakpoint",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "sizes": [Function],
      },
      "render": [Function],
      "styledComponentId": "csx__Breakpoint-sc-16ry0sm-0",
      "target": "div",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  sizes={
    Array [
      true,
      true,
      false,
      false,
      false,
    ]
  }
>
  Some text
</StyledComponent>
`;
